src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | object {
+  if (items.length === 0) {
```

Comment:
Returning an empty object for an empty input array is correct, but it might be more consistent to return an empty `GroupsMap<T>` instead of a plain object to maintain the return type consistency across all code paths.